ops/_tracing/hacks.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2025 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+# file except in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+"""Workarounds for various Juju bugs."""
+
+from __future__ import annotations
+
+import logging
+import os
+import shutil
+from collections import defaultdict
+from typing import Any
+
+import opentelemetry.trace
+from importlib_metadata import distributions  # type: ignore
+
+logger = logging.getLogger(__name__)
+tracer = opentelemetry.trace.get_tracer(__name__)
+
+
+# FIXME must this hack be run before OTEL packages are imported?
+# must test this
+@tracer.start_as_current_span('ops.remove_stale_otel_sdk_packages')
+def remove_stale_otel_sdk_packages() -> None:
```

Comment:
It seems awful that we have to do this. I don't suppose we're open to saying "you don't get tracing unless you're using Juju 3.5" or something like that? Or leaving this as a problem for charms to solve if they want to support older Juju?